frontend/packages/console-dynamic-plugin-sdk/src/extensions/console-types.ts

```diff
@@ -145,6 +146,32 @@ export type PrometheusResponse = {
   warnings?: string[];
 };
 
+export type PrometheusPollProps = {
+  /** Delay between polling requests */
+  delay?: number;
+  /** One of the well-defined Prometheus API endpoints */
+  endpoint: PrometheusEndpoint;
+  namespace?: string;
+  /** Prometheus query */
+  query: string;
```

Comment:
lets make query optional so we do not have to set it to empty string in case we dont have a query to watch yet

```suggestion
  query?: string;
```

Reply:
done